plugin/introspection/plugin.go

```diff
@@ -0,0 +1,96 @@
+package introspection
+
+import (
+	"github.com/99designs/gqlgen/codegen"
+	"github.com/99designs/gqlgen/codegen/config"
+	"github.com/99designs/gqlgen/codegen/templates"
+	"github.com/vektah/gqlparser/ast"
+)
+
+type Config struct {
+	Directives config.StringList `yaml:"directives"`
+}
+
+type Plugin struct {
+	cfg Config
+}
+
+func New(cfg Config) *Plugin {
+	return &Plugin{cfg: cfg}
+}
+
+func (p *Plugin) Name() string {
+	return "introspection"
+}
+
+func (p *Plugin) MutateConfig(cfg *config.Config) error {
+	for _, d := range p.cfg.Directives {
+		cfg.Directives[d] = config.DirectiveConfig{SkipRuntime: true}
+	}
+	return nil
+}
+
+func (p *Plugin) MutateSchema(schema *ast.Schema) error {
+	p.injectIntrospectionAuth(schema)
+	return nil
+}
+
+func (p *Plugin) GenerateCode(cfg *codegen.Data) error {
+	data := Data{
+		Data:       cfg,
+		Directives: nil,
+	}
+	for _, directiveName := range p.cfg.Directives {
+		for _, directive := range cfg.Schema.Directives {
+			if directive.Name == directiveName {
+				data.Directives = append(data.Directives, directive)
+			}
+		}
+	}
+	return templates.Render(templates.Options{
+		PackageName:     cfg.Config.Exec.Package,
+		Filename:        "introspection.go",
+		RegionTags:      true,
+		GeneratedHeader: true,
+		Data:            &data,
+		Funcs:           nil,
+	})
+}
+
+func (p *Plugin) injectIntrospectionAuth(schema *ast.Schema) {
+	introspection := &ast.DirectiveDefinition{
+		Description: "",
+		Name:        "introspection",
+		Arguments:   nil,
+		Locations: []ast.DirectiveLocation{
+			ast.LocationInputFieldDefinition,
+			ast.LocationFieldDefinition,
+		},
+		Position: nil,
+	}
```

Comment:
`@introspection` does not have an any (common) specs. I want to hide it in IntrospectionQuery.
I recommend add something like `SkipRuntime: true` or `Builtin: true` .
but wanna generate directives in code. 🤔 

@vektah do you have any opinion in this topic?

Reply:
Yep, it already exists https://github.com/99designs/gqlgen/blob/master/codegen/directive.go#L24